tests/glsl-intrinsic/shader-subgroup/shader-subgroup-ballot.slang

```diff
@@ -0,0 +1,127 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+// not testing cuda due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2 
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+
+
+// required since at the time of making, __BuiltinType does not allow simple casting, so we have a split test
+
+__generic<T : __BuiltinLogicalType>
+bool test1BroadcastX() {
+    return true
+        && subgroupBroadcast(T(1), 0) == T(1)
+        && subgroupBroadcastFirst(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinLogicalType, let N : int>
+bool testVBroadcastX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupBroadcast(gvec(T(1)), 0) == gvec(T(1))
+        && subgroupBroadcastFirst(gvec(T(1))) == gvec(T(1))
```

Comment:
This is a nitpick but I want to leave a quick comment; feel free to ignore.
The spec shows the signature of the function, subgroupBroadcast as following,
```
        genType  subgroupQuadBroadcast(genType value,  uint id);
        genIType subgroupQuadBroadcast(genIType value, uint id);
        genUType subgroupQuadBroadcast(genUType value, uint id);
        genBType subgroupQuadBroadcast(genBType value, uint id);
        genDType subgroupQuadBroadcast(genDType value, uint id);
```
You can imitate it as it is.
```
  typealias genType = vector<float,N>;
  typealias genIType = vector<int,N>;
  typealias genUType = vector<uint,N>;
  typealias genBType = vector<bool,N>;
  typealias genDType = vector<double,N>;
  return true
        && genType(1) == subgroupQuadBroadcast(genType(1),  uint(0))
        && genIType(1) == subgroupQuadBroadcast(genIType(1), uint(0))
        && genUType(1) == subgroupQuadBroadcast(genUType(1), uint(0))
        && genBType(1) == subgroupQuadBroadcast(genBType(1), uint(0))
        && genDType(1) == subgroupQuadBroadcast(genDType(1), uint(0))
```
It becomes little different but still very similar to how they appear on the document.

Reply:
I don't think I will adopt this for shader_subgroup specifically but I will for other ext's (I think it is a good idea todo when consistent). 

This can't be done to match almost exactly for some of the shader_subgroup functions (unless I do something like `genIType` and `genIVecType`, meaning there would be a consistency problem, possibly confusing since the code structure of the tests are already the same)